app/main/views/suppliers.py

```diff
@@ -86,6 +88,119 @@ def download_agreement_file(supplier_id, framework_slug, document_name):
     return redirect(url)
 
 
+@main.route('/suppliers/<supplier_id>/countersigned-agreements/<framework_slug>/'
+            '<document_name>', methods=['GET'])
+@login_required
+@role_required('admin-ccs-sourcing')
+def list_countersigned_agreement_file(supplier_id, framework_slug, document_name):
```

Comment:
They are there for archiving, so that we can manually roll back if someone really messes up. I think for the purpose of this interface we can probably assume there is only one. I'm not 100% sure on this though, it would be worth speaking to @cathrooney to see if they need to see the history of the file. In any case, the prefix we have here won't match those archive files as they start with the timestamp.